File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/casilla_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class CasillaImpuesto < Casilla
  def initialize(nombre,valor)
    super(nombre)
    @importe = valor
  end
  
  def recibeJugador(actual,todos)
    if(jugadorCorrecto(actual,todos))
      super(actual,todos)
      todos[actual].pagaImpuesto(@importe)
    end
  end
  
  def toString
    cadena="\nCasilla: #{@nombre}  Importe: #{@importe} \n\n"
    return cadena
  end
  public :toString
end
end

File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author pc
 */
import java.util.ArrayList;
public class CasillaImpuesto extends Casilla{
    private float importe;
     
     CasillaImpuesto(float cantidad,String nombre){
        super(nombre);
        importe=cantidad;
    }
     
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
       if(jugadorCorrecto(actual, todos)) {
        informe(actual, todos);
        todos.get(actual).pagaImpuesto(importe);
       }
    }
    @Override
    public String toString() {
        return "Casilla: "+this.getNombre() + "\n     Importe: " + importe ;
    }
}
